language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureHelpUtil.java

```diff
@@ -243,20 +201,34 @@ private SignatureHelpUtil() {
                     break;
                 }
                 tokenAcceptor.accept(commonToken.getType());
-                tokenText.add(commonToken.getText());
+                tokenTexts.add(commonToken.getText());
             }
         }
 
+        // Remove if any comma[,] when pendingRParenthesis is less than zero
+        int lastIndex = tokenTexts.size() - 1;
+        if (COMMA.equals(tokenTexts.get(lastIndex)) && pendingRParenthesisCount[0] <= 0) {
+            tokenTexts.remove(lastIndex);
+        }
+
+        //  Remove if any semi-colon[;]
+        lastIndex = tokenTexts.size() - 1;
+        if (SEMI_COLON.equals(tokenTexts.get(lastIndex))) {
+            tokenTexts.remove(lastIndex);
+        }
+
         // Add Missing Parenthesis
         while (pendingRParenthesisCount[0] < 0) {
-            tokenText.add(")");
+            tokenTexts.add(")");
             pendingRParenthesisCount[0]++;
         }
-        return tokenText;
+        return Pair.of(String.join("", tokenTexts), parameterOffset);
     }
 
     private static Optional<String> parseAndGetFunctionInvocationPath(String subRule, LSServiceOperationContext context)
             throws CompilationFailedException {
+        // Replace 'optional field access' with 'field access' to avoid empty top-level nodes
+        subRule = subRule.replaceAll("\\?.", ".");
```

Comment:
Can there be semantically inconsistent issues if we do this?

Reply:
This won't be an issue. Idea is to generate a path string seperated by dots[.] to resolve `BInvokableSymbol` out of an function invocation text. Here we treat Optional field access as legitimate field access to find the correct function `BInvokableSymbol` to generate the signature help text.